README.md

```diff
@@ -12,62 +51,164 @@ The recommended way to install is through Composer:
+    '{fakturoid-client-id}',
+    '{fakturoid-client-secret}',
+    'PHPlib <your@email.cz>'
+);
+$fManager->authClientCredentials();
+```
+
+## Processing credentials using the credentials callback:
+The way callback works is that the library calls the callback function whenever the credentials are changed. This is useful because the token is automatically refreshed after its expiration.
+```php
+$fManager->setCredentialsCallback(new class implements \Fakturoid\Auth\CredentialCallback {
+    public function __invoke(?\Fakturoid\Auth\Credentials $credentials = null): void
+    {
+        // Save credentials to database or another storage
+    }
+});
+```
+
 
 ## Usage
 
+### Upload credentials to the Fakturoid manager
+If you run a multi-tenant application or an application that processes documents in parallel, you need to set Credentials correctly. Each time a new access token is obtained, the previous one is invalidated. For these needs there is `AuthProvider::setCredentials()` and also `CredentialCallback`.
+```php
+$fManager = new \Fakturoid\FakturoidManager(
+    new ClientInterface(), // PSR-18 client
+    '{fakturoid-client-id}',
+    '{fakturoid-client-secret}',
+    'PHPlib <your@email.cz>'
+);
+// restore credentials from storage
+$credentials = new \Fakturoid\Auth\Credentials(
+    'refreshToken',
+    'accessToken',
+    (new DateTimeImmutable())->modify('-2 minutes'),
+    \Fakturoid\Enum\AuthTypeEnum::AUTHORIZATION_CODE_FLOW // or \Fakturoid\Enum\AuthTypeEnum:CLIENT_CREDENTIALS_CODE_FLOW
+);
+    
+$fManager->getAuthProvider()->setCredentials($credentials);
+$fManager->setCredentialsCallback(new class implements \Fakturoid\Auth\CredentialCallback {
+    public function __invoke(?\Fakturoid\Auth\Credentials $credentials = null): void
+    {
+        // Save credentials to database or another storage
+    }
+});
+```
+
+### Switch account
+
 ```php
-require_once '/path/to/lib/Fakturoid.php';
-$f = new Fakturoid\Client('..slug..', '..user@email.cz..', '..api_key..', 'PHPlib <your@email.cz>');
+$fManager = new \Fakturoid\FakturoidManager(
+    new ClientInterface(), // PSR-18 client
+    '{fakturoid-account-slug}',
+    '{fakturoid-client-id}',
+    '{fakturoid-client-secret}',
+    'PHPlib <your@email.cz>'
+);
+$fManager->authClientCredentials();
+$fManager->getSettingProvider()->listBankAccounts();
+
+// switch account and company    
+$fManager->setAccountSlug('{fakturoid-account-slug-another}');
+$fManager->getSettingProvider()->listBankAccounts();
+```
+
+### Basic usage
+```php
+require __DIR__ . '/vendor/autoload.php';
+$fManager = new \Fakturoid\FakturoidManager(
+    new ClientInterface(), // PSR-18 client
+    '{fakturoid-client-id}',
+    '{fakturoid-client-secret}',
+    'PHPlib <your@email.cz>'
+);
+$fManager->authClientCredentials();
+
+// get current account
+$account = $fManager->getSettingProvider()->getCurrentUser();
+$fManager->setAccountSlug($account->getBody()->default_account);
```

Comment:
Tohle bych změnil na `$user` a `get current user`, je to lehce zavádějící. Ještě si tipnu, že `default_account` bude ve většině případů null, tak co tam dát místo toho třeba totok: `$fManager->setAccountSlug($user->getBody()->accounts[0]->slug);`. Možná to je už zbytečný, ale to přejmenování bych udělal.